Sustainsys.Saml2/WebSSO/MetadataCommand.cs

```diff
@@ -33,18 +33,24 @@ public CommandResult Run(HttpRequestData request, IOptions options)
 
             var urls = new Saml2Urls(request, options);
 
-            var metadata = options.SPOptions.CreateMetadata(urls);
+            /* Dynamically fetch the identity provider for the entityid.
+             * This will ensure any settings controlled with a feature flag or dynamically refreshed values will be updated after Identity Server startup.
+             * We use these identity provider specific settings instead of the startup settings for the rest of the flow.
+             */
+            IdentityProvider identityProvider = options.Notifications.GetIdentityProvider(options.SPOptions.EntityId, null, options);
```

Comment:
Is the metadata request org-specific? Was just looking through our implementation of GetIdentityProvider and it doesn't look at the entityid and instead grabs the partition id from the header or query parameter. I think as long as the metadata request has that header due to going through routing, we should be good.

Reply:
Yes it is org specific. 

MSI url: https://localhost:44351/identity/Saml2?PartitionId=efd6ce42-779d-4bc9-a28e-a7934aae3736
Cloud and SF url: https://cloud.uipath.com/7c34aab7-605e-409c-9ad9-8d169c061243/identity_/Saml2